src/main/java/org/ieeervce/gatekeeper/controller/UserController.java

```diff
@@ -67,9 +69,10 @@ User addUser(@RequestBody UserDTO userDTO) throws InvalidDataException {
         return userService.saveUser(user);
     }
     @GetMapping
-    public String getUser()
+    public UserAuthDTO getUser()
     {
-        return getRequesterDetails();
+        User optionalUser = userService.getUserByEmail(getRequesterDetails()).get();
```

Comment:
Why not throw the exception we have, instead of an implicit NullPointerException

Reply:
Makes sense will do that,but i do wonder if its needed because the routes itself are protected and authorized,so if a person arrives at that point then wont the requester actually exist?

Reply:
Okay. Then I suggest updating the var name to just `user`, because it's not optional